File: api/schemas/legislative.py
# api/schemas/legislative.py
from __future__ import annotations
from pydantic import BaseModel, Field
from api.schemas.status import DataMode


class BoeItem(BaseModel):
    boe_no: str | None = None
    title: str
    section: str = ""
    department: str = ""
    date: str
    url: str | None = None
    type: str = ""
    relevance: str = "media"


class BoeResponse(BaseModel):
    items: list[BoeItem] = Field(default_factory=list)
    date: str
    mode: DataMode
    total: int


class Initiative(BaseModel):
    id: str
    title: str
    type: str = ""
    proponent: str = ""
    status: str = "Pendiente"
    submitted_at: str | None = None
    urgency: str = "low"  # low | medium | high


class InitiativesResponse(BaseModel):
    items: list[Initiative] = Field(default_factory=list)
    mode: DataMode
    total: int
    active: int
    critical: int


class LegislativeKpis(BaseModel):
    active_initiatives: int
    approved_this_month: int
    critical_tramitation: int
    upcoming_votes: int
    mode: DataMode

File: apps/web/lib/types/legislative.ts
// apps/web/lib/types/legislative.ts
import type { DataMode } from "./status";

export interface BoeItem {
  boe_no: string | null;
  title: string;
  section: string;
  department: string;
  date: string;
  url: string | null;
  type: string;
  relevance: string;
}

export interface BoeResponse {
  items: BoeItem[];
  date: string;
  mode: DataMode;
  total: number;
}

export interface Initiative {
  id: string;
  title: string;
  type: string;
  proponent: string;
  status: string;
  submitted_at: string | null;
  urgency: "low" | "medium" | "high";
}

export interface InitiativesResponse {
  items: Initiative[];
  mode: DataMode;
  total: number;
  active: number;
  critical: number;
}

export interface LegislativeKpis {
  active_initiatives: number;
  approved_this_month: number;
  critical_tramitation: number;
  upcoming_votes: number;
  mode: DataMode;
}
